### PR TITLE
Add context help

### PR DESCRIPTION
**Pull request by csjall**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/niveristand-routing-and-faulting-custom-device/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

This PR adds context help to the Routing and Faulting, SLSC Switch System Explorer pages. I used snippets from the [User Guide](https://github.com/ni/niveristand-routing-and-faulting-custom-device/blob/master/Docs/User%20Guide.md) for the content.

image

image

image

### Why should this Pull Request be merged?

Replace boilerplate documentation with useful documentation.

### What testing has been done?

Built custom devices and viewed in VeriStand.

### CODE REVIEW

File: Source/Routing and Faulting/Help/Build CHM.log [hunk 1]
```diff
@@ -1,28 +1,2 @@
...
-0	Graphics
-
-
-
-
-
-Created c:\Git\niveristand-routing-and-faulting-custom-device\Source\Routing and Faulting\Help\Routing and Faulting.chm, 80,176 bytes
-
-Compression decreased file by 250,365 bytes.
+HHC5010: Error: Cannot open "c:\Git\niveristand-routing-and-faulting-custom-device\Source\Routing and Faulting\Help\Routing and Faulting.chm". Compilation stopped.
```
Comment (csjall) [inline Source/Routing and Faulting/Help/Build CHM.log]: You cannot build a CHM file if the file is open.

File: Source/SLSC Switch/Help/HTML Help Source/Main.htm [hunk 2]
```diff
@@ -14,13 +14,44 @@
...
+		</style>
 	</head>
 	<body>
 		<noscript>
 			<p class="Body">JavaScript is disabled. <a href="veristandmerge.chm::/JavaScript_Disabled.html">Details</a></p>
 			<hr width="100%" noshade>
 		</noscript>
 		<h1>SLSC Switch</h1>
-		Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</br>
+		<p class="Body">The <strong>SLSC Switch Custom Device</strong> is a VeriStand add-on that supports SLSC routing modules.</p>
```
Comment (rtzoeller) [inline Source/SLSC Switch/Help/HTML Help Source/Main.htm]: > `The <strong>SLSC Switch Custom Device</strong> is a VeriStand add-on`

This feels unnecessary in this context. We are showing the help in VeriStand.

File: Source/Routing and Faulting/Help/HTML Help Source/Main.htm [hunk 3]
```diff
@@ -21,6 +21,9 @@
...
+		<p class="Body">The <strong>Routing and Faulting Custom Device</strong> is a VeriStand add-on that configures aliases and routing channels.</p>
+		<h2>What do you want to do?</h2>
+		<h3>Script Routing and Faulting Configuration</h3>
+		<p class="Body">Open the scripting examples on disk from <strong>Source/Scripting Examples/Scripting Examples.lvproj</strong>. Run <strong>Configure Routing and Faulting Custom Device.vi</strong> to add routing channels, states, and connections.</p>
```
Comment (rtzoeller) [inline Source/Routing and Faulting/Help/HTML Help Source/Main.htm]: > `<strong>Source/Scripting Examples/Scripting Examples.lvproj</strong>`

Should this be the installed location (TBD)?

File: Source/Routing and Faulting/Help/HTML Help Source/Routing Channels.htm [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+		<script src="launchhelp.js" type="text/javascript"></script><script src="js/dynamicjumps.js" type="text/javascript"></script>
+	</head>
+	<body>
+		<noscript>
+			<p class="Body">JavaScript is disabled. <a href="veristandmerge.chm::/JavaScript_Disabled.html">Details</a></p>
+			<hr width="100%" noshade>
+		</noscript>
+		<h1>Routing Channels</h1>
+		<p class="Body">By default, routing channels are not added. Add routing channels for each controllable collection of connections.</p>
+		<p class="Body">Each routing channel contains one or more user-defined states. Common states for a fault injection application include <strong>Passthrough</strong>, <strong>Open Circuit</strong>, <strong>Short to Ground</strong>, <strong>Short to Power</strong>, and <strong>Pin-to-Pin Short</strong>. By default, the <strong>Open Circuit</strong> state is added. Add additional states to match the requirements of the fault injection application.</p>
```
[the remark this answers is not in the document]
Comment (csjall) [reply Source/Routing and Faulting/Help/HTML Help Source/Routing Channels.htm]: This text is copied from the User Guide. If we want to revisit the terminology, then we should updated both documents.

File: Source/Routing and Faulting/Help/HTML Help Source/Routing Channels.htm [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+		<p class="Body">Each state contains a collection of connections. Each connection consists of a source and destination endpoint on a routing module. By default, no connections are added. Add connections to establish connections between the DUT and the I/O. A single connection cannot span multiple routing modules. Define multiple connections, one for each routing module, to connect a signal spanning multiple routing modules.</p>
```
[the remark this answers is not in the document]
Comment (csjall) [reply Source/Routing and Faulting/Help/HTML Help Source/Routing Channels.htm]: I will let @WillCawley weight in on wording.

File: Source/SLSC Switch/Help/HTML Help Source/Main.htm (hunk 2 above)
Comment (csjall) [reply Source/SLSC Switch/Help/HTML Help Source/Main.htm]: Most of our custom devices start with this rather obvious introduction.

- The **Scan Engine and EtherCAT** custom device allows you to communicate with I/O in local and remote chassis using the NI Scan Engine.
- This **SLSC EDS Addon** provides the ability to monitor and communicate with any SLSC Module inside a SLSC Chassis in a generic way.
- This **SLSC-12201 Module Addon** provides the ability to monitor and communicate with SLSC-12201 Digital Input/Output Module in a SLSC Chassis.

We can omit "VeriStand" and maybe be consist with our spelling of addon(add-on?)(add on?).

File: Source/Routing and Faulting/Help/HTML Help Source/Main.htm (hunk 3 above)
Comment (csjall) [reply Source/Routing and Faulting/Help/HTML Help Source/Main.htm]: Let's remove the folder location and just mention the example name. Once this is in the example database people should able to find it quickly.

File: Source/Routing and Faulting/Help/HTML Help Source/Routing Channels.htm [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+		<link rel="STYLESHEET" type="text/css" href="css/expandable_tree.css">
+		<script src="launchhelp.js" type="text/javascript"></script><script src="js/dynamicjumps.js" type="text/javascript"></script>
+	</head>
+	<body>
+		<noscript>
+			<p class="Body">JavaScript is disabled. <a href="veristandmerge.chm::/JavaScript_Disabled.html">Details</a></p>
+			<hr width="100%" noshade>
+		</noscript>
+		<h1>Routing Channels</h1>
+		<p class="Body">By default, routing channels are not added. Add routing channels for each controllable collection of connections.</p>
```
[the remark this answers is not in the document]
Comment (csjall) [reply Source/Routing and Faulting/Help/HTML Help Source/Routing Channels.htm]: I will revisit this page and make it more specific.

File: Source/Routing and Faulting/Help/HTML Help Source/Main.htm [hunk 7]
```diff
@@ -21,6 +21,9 @@
 			<hr width="100%" noshade>
 		</noscript>
 		<h1>Routing and Faulting</h1>
-      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Tristique et egestas quis ipsum suspendisse ultrices gravida dictum fusce. Laoreet non curabitur gravida arcu ac tortor dignissim. Enim praesent elementum facilisis leo. Viverra mauris in aliquam sem. Tristique et egestas quis ipsum. Et egestas quis ipsum suspendisse ultrices gravida. Imperdiet nulla malesuada pellentesque elit eget gravida cum sociis natoque. Vitae congue eu consequat ac felis donec. Dictum sit amet justo donec enim diam vulputate. Semper auctor neque vitae tempus quam. Quis eleifend quam adipiscing vitae proin. Mollis nunc sed id semper. Lorem ipsum dolor sit amet consectetur adipiscing elit. Risus ultricies tristique nulla aliquet enim tortor at.</br>
+		<p class="Body">The <strong>Routing and Faulting Custom Device</strong> is a VeriStand add-on that configures aliases and routing channels.</p>
```
Comment (jashnani) [inline Source/Routing and Faulting/Help/HTML Help Source/Main.htm]: configures endpoint aliases

File: Source/Routing and Faulting/Help/HTML Help Source/Main.htm [hunk 8]
```diff
@@ -21,6 +21,9 @@
 			<hr width="100%" noshade>
 		</noscript>
 		<h1>Routing and Faulting</h1>
-      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Tristique et egestas quis ipsum suspendisse ultrices gravida dictum fusce. Laoreet non curabitur gravida arcu ac tortor dignissim. Enim praesent elementum facilisis leo. Viverra mauris in aliquam sem. Tristique et egestas quis ipsum. Et egestas quis ipsum suspendisse ultrices gravida. Imperdiet nulla malesuada pellentesque elit eget gravida cum sociis natoque. Vitae congue eu consequat ac felis donec. Dictum sit amet justo donec enim diam vulputate. Semper auctor neque vitae tempus quam. Quis eleifend quam adipiscing vitae proin. Mollis nunc sed id semper. Lorem ipsum dolor sit amet consectetur adipiscing elit. Risus ultricies tristique nulla aliquet enim tortor at.</br>
```
Comment (jashnani) [inline Source/Routing and Faulting/Help/HTML Help Source/Main.htm]: so sad to see "lorem ipsum dolor..." go away :/

File: Source/Routing and Faulting/Help/HTML Help Source/Routing Channels.htm (hunk 4 above)
[the remark this answers is not in the document]
Comment (jashnani) [reply Source/Routing and Faulting/Help/HTML Help Source/Routing Channels.htm]: I suspect that EES port uses that word. If so, we could leave it as-is.

[the remark this answers is not in the document]
Comment (csjall) [reply Source/Routing and Faulting/Help/HTML Help Source/Routing Channels.htm]: Here are some of the error categories:

**Pull request merged**